OneCommunityGlobal/HGNRest
# Juhitha took over for Jiayu & Shuhua - All Inventory Types page functionalities

# Description

Implement the functionalities of the All Inventory Types page, which the UI was implemented in this [PR](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1868)

## Related PRS (if any):

To test this backend PR you need to checkout the [#3833](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/3833) frontend PR.

## Main changes explained:

- Added missing PUT routes for updating inventory types (Reusables, Equipments, Consumables, Tools)
- Added missing DELETE routes for removing inventory types across all categories
- Fixed POST route for Reusables inventory type (was completely missing)
- Added missing POST/DELETE routes for inventory units management
- Updated bmInventoryTypeRouter.js with comprehensive CRUD operations for all inventory types
- Integrated controller methods for updateSingleInvType, deleteSingleInvType, addInvUnit, deleteInvUnit

## How to test:

1. Check into current branch Shuhua-Types_List_Page_Edit_Delete
2. Do npm install && npm start to run this PR locally on port 4500
3. Checkout the frontend [cleanup branch](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/3833)
4. Verify all inventory type operations work:
  - GET: Fetch Materials, Consumables, Tools, Equipment, Reusables
  - POST: Add new inventory types for all categories
  - PUT: Update existing inventory types
  - DELETE: Remove inventory types
